qcodes/instrument_drivers/Keysight/N52xx.py

```diff
@@ -377,12 +424,15 @@ def traces(self) -> ChannelList:
         """
         Update channel list with active traces and return the new list
         """
+        active_trace = self.active_trace()
         parlist = self.ask("CALC:PAR:CAT:EXT?").strip('"').split(",")
         self._traces.clear()
-        for trace in parlist[::2]:
-            trnum = PNATrace.parse_paramstring(trace)[2]
-            pna_trace = PNATrace(self, "tr{}".format(trnum), int(trnum))
+        for trace_name in parlist[::2]:
+            self.write(f"CALC:PAR:SEL '{trace_name}'")
+            trace_num = int(self.ask("CALC:PAR:MNUM?"))
```

Comment:
A general comment: could you make separate methods out of `.ask(<CMD>)` and `.write(<CMD>)` in the instrument class instead of explicitly using `.ask()` and `.write()` with explicit VISA commands in the trace subclass? This will allow reuse of these commands, reduce code duplication, and potentially reduce possible errors.

Reply:
Agreed, most direct `write`/`ask` calls are now removed.